doc/user-manual.md

```diff
@@ -1,216 +1,457 @@
-- [Metric Detail](#Metric-Detail)
-- [Metric Field](#Metric-Field)
-  - [Http Request](#Http-Request)
-  - [JDBC Statement](#JDBC-Statement)
-  - [JDBC Connection](#JDBC-Connection)
-  - [JVM Memory](#JVM-Memory)
-  - [JVM GC](#JVM-GC)
-  - [Kafka Client](#Kafka-Client)
-  - [RabbitMq Producer](#RabbitMq-Producer)
-  - [RabbitMq Consumer](#RabbitMq-Consumer)
-
-## Metric Detail
-
-The metric name's rule are as follows:
-
- Metric | Metric Type| 
----|---|
-Timers | 0
-Histograms | 1
-Meters | 2
-Counters | 3
-Gauges | 4
-
- Key Type | Metric Sub Type| Description |
----|---|---|
-SUCCESS | 00 | Method invoke d not exception 
-ERROR | 01 | Method call has error or http response 500, etc
-CHANNEL | 02 | RabbitMq publish message to channel or queue
-CONSUMER | 03 | RabbitMq or Kafka consume message
-PRODUCER | 04 | Kafka send message to topic
-CONSUMER_ERROR | 05 | RabbitMq or Kafka consume message has error
-PRODUCER_ERROR | 06 | Kafka send message has error
+  
+# User Manual
 
+- [User Manual](#user-manual)
+  - [Agent.properties](#agentproperties)
+    - [Getting the configuration file](#getting-the-configuration-file)
+    - [Internal HTTP Server](#internal-http-server)
+    - [Metric](#metric)
+    - [Kafka](#kafka)
+    - [HTTP Reqeust Metric](#http-reqeust-metric)
+    - [JDBC SQL Metric](#jdbc-sql-metric)
+    - [JDBC Connection Metric](#jdbc-connection-metric)
+    - [RabbitMQ Metric](#rabbitmq-metric)
+    - [Kafka Metric](#kafka-metric)
+    - [Redis Metric](#redis-metric)
+    - [JVM GC Metric](#jvm-gc-metric)
+    - [JVM Memory Metric](#jvm-memory-metric)
+    - [SQL MD5Dictionary](#sql-md5dictionary)
+    - [Tracing](#tracing)
+    - [Logging](#logging)
+  - [Prometheus Support](#prometheus-support)
+  - [Health Check and Readiness Check Endpoint](#health-check-and-readiness-check-endpoint)
+  - [Tracing](#tracing-1)
+    - [Tracing Component](#tracing-component)
+    - [Custom Span Tag](#custom-span-tag)
+      - [JDBC](#jdbc)
+      - [Cache](#cache)
+      - [RabbitMQ Producer And Consumer](#rabbitmq-producer-and-consumer)
+      - [Kafka Producer And Consumer](#kafka-producer-and-consumer)
+  - [Metric](#metric-1)
+    - [Metric Field](#metric-field)
+      - [HTTP Request](#http-request)
+      - [JDBC Statement](#jdbc-statement)
+      - [JDBC Connection](#jdbc-connection)
+      - [JVM Memory](#jvm-memory)
+      - [JVM GC](#jvm-gc)
+      - [Kafka Client](#kafka-client)
+      - [RabbitMq Producer](#rabbitmq-producer)
+      - [RabbitMq Consumer](#rabbitmq-consumer)
+  
+## Agent.properties
+`EaseAgent` uses the `report` module to control the tracing and metric collection behavior of each component. `agent.properties` is configured with various parameters of `report`. Changing these parameters can change the collection behavior. These parameters include: collection frequency, target queue, switch and other settings. Users can customize parameter.
+
+### Getting the configuration file
+Extracting the default configuration file
```

Comment:
you may extract default configuration from the JAR file or create new properties from a blank file